rocky/tools/forms/upload_raw.py

```diff
@@ -22,7 +30,31 @@ class UploadRawForm(forms.Form):
         allow_empty_file=False,
     )
 
+    ooi_id = forms.CharField(
+        label="Scan OOI",
+        widget=DataListInput(attrs={"placeholder": _("Click to select one of the available options")}),
+    )
+
     def clean_mime_types(self) -> set[str]:
         mime_types = self.cleaned_data["mime_types"]
 
         return {mime_type.strip() for mime_type in mime_types.split(",") if mime_type.strip()}
+
+    def __init__(
+        self,
+        connector: OctopoesAPIConnector,
+        ooi_list: list[tuple[str, str]],
+        *args,
+        **kwargs,
+    ):
+        self.octopoes_connector = connector
+        super().__init__(*args, **kwargs)
+        self.set_choices_for_widget("ooi_id", ooi_list)
+
+    def clean_ooi_id(self):
+        try:
+            data = self.cleaned_data["ooi_id"]
+            self.octopoes_connector.get(Reference.from_str(data), datetime.now(timezone.utc))
```

Comment:
Are we loading the OOI twice? Once when cleaning the form (and validating that the OOI Exists, and then another time when we want to find its primary-key, which we already had on this spot / from the original form input?

Reply:
You are absolutely right, sharp. I fixed that now.